### PR TITLE
Zpi 78 More tests for cluster functions

### PR DESCRIPTION
**Pull request by Ninja00Shadow**
I've added more tests, primarily for extraction functions. This should be enough for unit testing. Creating more tests will require significant changes and can do more harm than good.

### CODE REVIEW

File: backend/cluster/list_resources.go [hunk 1]
```diff
@@ -398,8 +430,9 @@ func extractKeys(resource unstructured.Unstructured, resourceType string, resour
 			}
 		}
 
-		resourceDetailsTruncated.Keys = fmt.Sprintf("%v", keys)
+		sort.Strings(keys)
```
Comment (samuelus) [inline backend/cluster/list_resources.go:433]: Does it need to be sorted?

Comment (samuelus) [reply backend/cluster/list_resources.go:433]: *general comment

Comment (Ninja00Shadow) [reply backend/cluster/list_resources.go:433]: Yes and no. Keys in map are non-deterministic. They change, which is probably not good in general, but trigers errors in tests quite often.

File: docs/test_resource_yamls/README.md [hunk 2]
```diff
@@ -8,11 +8,8 @@ The files are named according to the endpoint they were used to test. Some may c
 
 - `configmap.yaml`
 - `daemonset.yaml`
-- `deployment.yaml`
```
Comment (Nightbaka) [inline docs/test_resource_yamls/README.md:11]: why?

Comment (Ninja00Shadow) [reply docs/test_resource_yamls/README.md:11]: I removed files that are no longer needed. Installing the Helm chart now creates the ingress, deployment, and service automatically.

File: backend/cluster/list_resources_test.go [hunk 3]
```diff
@@ -0,0 +1,3426 @@
...
+	})
+}
+
+func TestExtractActive(t *testing.T) {
+	tests := []struct {
+		name                      string
+		resource                  unstructured.Unstructured
+		resourceType              string
+		expectedActive            string
+		transposedColumnsOverride map[string][]string
```
Comment (Nightbaka) [inline backend/cluster/list_resources_test.go:119]:
```suggestion
```
it is unused

**Review by Nightbaka — APPROVED** [review]
Honestly I haven't read all the tests, but from the ones I looked it looks like you covered the funcs pretty damn well. The sort changes are fine, as thats what frontend would like to do propably anyways. I've sneaked in commit deleting unused struct field as revenge for sneaking in deleting my unused database resources

Heh ~ Yours truly

**Review by samuelus — APPROVED** [review]

**Pull request merged**